hypershift-operator/controllers/platform/gcp/privateserviceconnect_controller.go

```diff
@@ -0,0 +1,592 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/support/upsert"
+	supportutil "github.com/openshift/hypershift/support/util"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/go-logr/logr"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/option"
+)
+
+const (
+	finalizer = "hypershift.openshift.io/gcp-private-service-connect"
+)
+
+// RBAC permissions for GCPPrivateServiceConnect controller
+// +kubebuilder:rbac:groups=hypershift.openshift.io,resources=gcpprivateserviceconnects,verbs=get;list;watch;update;patch
```

Comment:
nit: these are not used elsewhere and might give the impression that they cause some code to be generated. They're great as documentation though. I would just addd a note saying that they are for doc only.

Reply:
Added a comment about the documentation only purpose.